src/lib/schematics/utils/build-component.ts

```diff
@@ -168,9 +168,7 @@ export function buildComponent(options: ComponentOptions,
     const schematicFilesPath = resolve(dirname(context.schematic.description.path),
         schematicFilesUrl);
 
-    if (!options.styleext) {
-      options.styleext = determineDefaultStyleExt(project);
-    }
+    setDefaultSchematicOptions(project, options);
```

Comment:
@jelbourn I'm very unhappy with that function name. Do you have any better ideas?

Reply:
How about
```ts
options = {...getDefaultComponentOptions(project), ...options};
```

Reply:
That would have been perfect. Unfortunately the options will look something like: `{spec: undefined}` and therefore wouldn't be replaced by the default component options.

I've went with something like that:

```ts

// Add the default component option values to the options if an option is not explicitly
// specified but a default component option is available.
Object.keys(options)
  .filter(optionName => options[optionName] == null && defaultComponentOptions[optionName])
  .forEach(optionName => options[optionName] = defaultComponentOptions[optionName]);
```